### ISSUE TITLE
b2 suddenly fails to authorise

### ISSUE THREAD

**Issue by marrie** [ncw/rclone#788]
Cloud storage in question: b2
Latest rclone1.33
Mac os sierra 64 bit

Basically I went to start a script today noticing that it had just halted. I got an error in my log that there was an authorization error.

> 2016/10/13 22:07:17 Failed to create file system for "b2:samusic/4music": failed to authorize account: 
> failed to authenticate: Get https://api.backblazeb2.com/b2api/v1/b2_authorize_account: 
> x509: certificate signed by unknown authority
>

I can attach a log however that's all that's there. I have no clue what I broke this time, but it was working last night. I even tried away from my proxy and that didn't work.

Thanks.

**Comment by marrie** [comment]
By the way, I'm hoping that this issue will just fix itself some how. Maybe b2 is just broken, and they can do what ever they need to do.

**Comment by ncw** [comment]
`x509: certificate signed by unknown authority` means that there is something wrong with the SSL certificates.

It is possible onedrive messed up temporarily, or maybe there is something wrong with the certificates on your computer?

What happens when you go to https://api.onedrive.com/ - you should receive a JSON error message - if you get a certificate warning error from your computer then something is up with it.

It might be something to do with this: https://twitter.com/globalsign/status/786505261842247680?lang=da

**Comment by ncw** [comment]
Sorry just read that it was b2 not onedrive!

Try https://api.backblazeb2.com/b2api/v1/b2_authorize_account in your browser.  You should get a message about a missing token, not a security error.

**Comment by marrie** [comment]
I don't use 1drive. Actually. I don't even have a configuration for 1drive set up. I hope my configure thing didn't get corrupted or something as I didn't add any 1drive stuff as far as I know. I remember putting my b2 remote in the path. This is the same b2 script I ran a few nights ago with success. By the way I did get a jSun error.

{"error":{"code":"invalidRequest","message":"[path.String] A value is required but was not present in the request."}}

Ok, now what? Looks like my computer so far is fine. Knock on wood.

**Comment by marrie** [comment]
Attached is the script that worked up to last night. I edited out the folder paths.

#!/bin/bash


~/rclone/rclone \
sync /Volumes/drive/path/to/folder \
b2:path/to/bucket/and/folder \
--transfers 3 \
--low-level-retries 1 \
--retries 20 \
--exclude '.ds_store' \
-q 2>&1 | split -b 2097152 - /tmp/outputmusic.log

As you can see 1drive should not have appeared as I used b2 which is my shorthand for backblaze b2.

hope that helps.

**Comment by ncw** [comment]
Can you try this url please?  https://api.backblazeb2.com/b2api/v1/b2_authorize_account

**Comment by marrie** [comment]
Ok, I get this in my browser.

{
  "code": "missing_auth_token",
  "message": "Missing authorization token",
  "status": 401
}

Is that what I'm supposed to get? I'm getting a lot of expired messages  As of I think 2 nights ago. See a bug I replied to prior to this one.

Thanks.

**Comment by marrie** [comment]
Ok, I killed the script as there were a lot of expired token messages. I then restarted the script and got the following from the log.

2016/10/17 09:37:05 Failed to create file system for "b2:samusic/3archives": failed to authorize account: failed to authenticate: Get https://api.backblazeb2.com/b2api/v1/b2_authorize_account: x509: certificate signed by unknown authority


Is there somethingn I need to do on backblaze's site to fix this? I don't think I broke it this time. I don't remember doing anything to cause this.

Thanks.
